tgui/packages/tgui/interfaces/LanguageMenu.js

```diff
@@ -67,25 +91,50 @@ export const LanguageMenu = (props, context) => {
               />
             }>
             <LabeledList>
-              {unknown_languages.map((language) => (
-                <LabeledList.Item
-                  key={language.name}
-                  label={language.name}
-                  buttons={
-                    <Button
-                      content="Grant"
-                      onClick={() =>
-                        act('grant_language', {
-                          language_name: language.name,
-                        })
+              {unknown_languages.map((each_language, index) => {
+                const languageData = language_static_data[each_language.name];
+                return (
+                  <>
+                    <LabeledList.Item
+                      key={languageData.name}
+                      label={
+                        <Stack vertical mb={-3.5} ml={1}>
+                          <Stack.Item>
+                            <Stack mb={-1}>
+                              <Stack.Item mr={0.2}>
+                                <Box className={classes(['chat16x16', 'language-' + languageData.icon_state])} />
+                              </Stack.Item>
+                              <Stack.Item>{languageData.name}</Stack.Item>
+                            </Stack>
+                          </Stack.Item>
+                          <Stack.Item>
+                            <Box pl={4}>Chat key: ,{languageData.key}</Box>
+                          </Stack.Item>
+                        </Stack>
                       }
-                    />
-                  }>
-                  {language.desc} Key: ,{language.key} {!!language.shadow && '(gained from mob)'}{' '}
-                  {language.can_understand ? 'Can understand.' : 'Cannot understand.'}{' '}
-                  {language.can_speak ? 'Can speak.' : 'Cannot speak.'}
-                </LabeledList.Item>
-              ))}
+                      buttons={
+                        <Button
+                          content="Grant"
+                          onClick={() =>
+                            act('grant_language', {
+                              language_name: languageData.name,
+                            })
+                          }
+                        />
+                      }>
+                      <Box>{languageData.desc}</Box>
+                      <Box pl={4}>
+                        Chat key: ,{languageData.key} {!!each_language.shadow && '(Language gained from mob)'}
+                      </Box>
```

Comment:
Why do we need this here if we have it just above?

Reply:
new players will not notice if there is a shortcut key for a language when they read a wall of texts.

Reply:
oh wait. it's double.